rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/CoordinatorBookKeeper.java

```diff
@@ -35,43 +36,101 @@
   // Cache to store hostnames of live workers in the cluster.
   protected Cache<String, Boolean> liveWorkerCache;
 
+  // Cache to store hostnames of caching-validated worker nodes.
+  protected Cache<String, Boolean> cachingValidatedWorkerCache;
+
+  // Cache to store hostnames of file-validated worker nodes.
+  protected Cache<String, Boolean> fileValidatedWorkerCache;
+
   public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoundException
   {
     this(conf, metrics, Ticker.systemTicker());
   }
 
   @VisibleForTesting
-  CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws FileNotFoundException
+  public CoordinatorBookKeeper(Configuration conf, MetricRegistry metrics, Ticker ticker) throws FileNotFoundException
   {
     super(conf, metrics, ticker);
-    this.liveWorkerCache = CacheBuilder.newBuilder()
-        .ticker(ticker)
-        .expireAfterWrite(CacheConfig.getWorkerLivenessExpiry(conf), TimeUnit.MILLISECONDS)
-        .build();
+    this.liveWorkerCache = createHealthCache(conf, ticker);
```

Comment:
Why we are invoking `createHealthCache` three times with the same arguments ?

Reply:
Since the live worker and validation caches all share the same settings, this centralizes the creation logic.
I considered using just one cache for storing all the validation results, but this would require more processing to determine the number of validated workers when reporting metrics values.